cpg-core/src/main/java/de/fraunhofer/aisec/cpg/frontends/java/ExpressionHandler.java

```diff
@@ -559,7 +569,7 @@ private de.fraunhofer.aisec.cpg.graph.statements.expressions.Expression handleNa
 
       var recordDeclaration = this.frontend.getScopeManager().getCurrentRecord();
 
-      if (recordDeclaration != null && Objects.equals(recordDeclaration.getName(), name)) {
+      if (recordDeclaration != null && recordDeclaration.getName().endsWith(name)) {
```

Comment:
Isn't such a comparison a problem when we have a record with name `Bla` and another with the name `MuchBla`?

Reply:
Maybe the name of the function `endsWith` is misleading. It basically checks if the end of the name "path" is the same. For example `A::B` and `B` will match (if the delimiter is `::`). `MuchBla` and `Bla` will not match.

Do you have an idea for another name of this function?

Reply:
How about `lastPartMatches`?

Reply:
Yeah I have seen it during the review further down the line, did not know that you already get the comments here before I submit the review. I just got confused with the string `endsWith` maybe `isNamePrefixOf` or similar if we want to be more precise.

Reply:
Äh not prefixed, thats the wrong one

Reply:
Ok, I renamed the affected methods to `lastPartsMatch`